### ISSUE TITLE
Change event fire twice when clicking on a switch

### ISSUE THREAD

**Issue by jakwarrior** [OnsenUI/OnsenUI#1503]
[](Github issues should be used only for bugs and feature requests. For questions please visit https://stackoverflow.com/tags/onsen-ui or https://community.onsen.io. For bugs please use the following template.)

__Environment__
Onsen 2 RC 15
Angular 1.5.5
Google Chrome 51

__Encountered problem__
I juste found a weird bug. When I click on a switch with Chrome and the device simulator enabled, the "change" event fire twice and the switch doesn't change its position. But it works fine directly in the browser or a real device.

__How to reproduce__
[](If possible provide a working example where the issue is reproduced. For this, go to http://tutorial.onsen.io/ , select a framework and a template and modify or replace it. This will use the latest Onsen UI version released. After you finish, click 'Export to Codepen' button, save it and link it here.)
Go to [https://codepen.io/anon/pen/YWxYzy](url) with Chrome, enable the device mode ([https://developers.google.com/web/tools/chrome-devtools/iterate/device-mode/](url)) and choose Nexus 7 as device, click on the switch => the event fire twice and the switch doesn't change its position.

**Comment by jakwarrior** [comment]
Thank you for your detailed answer. I just found this bug is not limited to the browser's device mode. I can reproduce the bug on a touch device with Windows as OS and Chrome (or Chromium). Here are all my tests (with [https://codepen.io/anon/pen/YWxYzy](https://codepen.io/anon/pen/YWxYzy):

-  Desktop device with Windows and several browsers (Chrome, Firefox, IE11) => OK
- Windows Touch Device (Surface Pro 3 with W8.1) => OK with Firefox and IE11 but KO with Chrome and Chromium
- iOS and Android => OK

So I think there is still a bug only when Onsen is used with a Windows tablet and Chrome.

**Comment by IliaSky** [comment]
Hmm... Well I am guessing since it's the same behaviour as the previous case probably the reason is also the same - chrome sending both `click` and `tap` events for only one action. Unfortunately testing this on our side won't be really easy and the most likely fix would be something like a 10ms delay with a check, which is not very elegant. So I think we may keep this issue closed for now since it seems like a low priority since the case is relatively rare. If you're interested in a fix on your side I would be willing to help.

**Comment by argelius** [comment]
@jakwarrior @IliaSky I agree. It looks like the device is generating two events in Chrome. Maybe we can try updating the version of Hammer that is bundled in Onsen UI.

However, there is no way for us to verify that it has been fixed since we don't have access to a Surface device.

**Comment by jakwarrior** [comment]
@argelius You can use Chrome's device mode to test, like explained in my first post. No need of a Surface device.
I will try to fix this on my side in the meantime.
